dvc/remote/gdrive/client.py

```diff
@@ -0,0 +1,284 @@
+from time import sleep
+import logging
+import posixpath
+import os
+
+from funcy import cached_property
+
+from requests import ConnectionError
+
+from dvc.progress import progress
+from dvc.remote.gdrive.utils import (
+    metadata_isdir,
+    response_is_ratelimit,
+    MIME_GOOGLE_APPS_FOLDER,
+    response_error_message,
+)
+from dvc.remote.gdrive.exceptions import (
+    GDriveError,
+    GDriveHTTPError,
+    GDriveResourceNotFound,
+)
+from dvc.remote.gdrive.oauth2 import OAuth2
+
+logger = logging.getLogger(__name__)
+
+
+class GDriveClient:
+
+    GOOGLEAPIS_BASE_URL = "https://www.googleapis.com/"
+    TIMEOUT = (5, 60)
+
+    def __init__(
+        self,
```

Comment:
use kwargs instead of a long list here?

Reply:
Hm. I'd better pass the OAuth2 instance. And this method also needs a docstring, probably.